tempesta_fw/http.c

```diff
@@ -516,18 +516,20 @@ tfw_http_adjust_resp(TfwHttpResp *resp, TfwHttpReq *req)
 static void
 tfw_http_req_cache_cb(TfwHttpReq *req, TfwHttpResp *resp)
 {
-	int r;
+	int r, req_flags = req->flags;
+	void *req_sk = req->conn->sk;
 	TfwConnection *conn;
 
 	if (resp) {
 		/*
-		 * We have prepared response, send it as is.
-		 * The response is pass through ot was generated from
-		 * the cache, so unrefer all its data.
+		 * The response is prepared, send it as is. The response
+		 * is either passed through from the back-end server, or
+		 * it is generated from the cache, so unrefer all its data.
 		 */
-		if (tfw_http_adjust_resp(resp, req))
-			return;
-		tfw_connection_send(req->conn, (TfwMsg *)resp, true);
+		if (tfw_http_adjust_resp(resp, req) == 0)
+			tfw_connection_send(req->conn, (TfwMsg *)resp, true);
+		if (req_flags & TFW_HTTP_CONN_CLOSE)
+			ss_droplink(req_sk);
```

Comment:
Hm, shouldn't we do ```tfw_http_conn_msg_free((TfwHttpMsg *)req)``` here? It seems the job is done and we don't see where caller frees ```req```....